diff --git a/src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java b/src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java
index 3677f0071e..7c116ae93e 100644
--- a/src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java
+++ b/src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java
@@ -1380,8 +1380,6 @@ public class CppCompileAction extends AbstractAction implements IncludeScannable
     fp.addBytes(commandLineKey);
     fp.addBoolean(validateTopLevelHeaderInclusions);
 
-    actionKeyContext.addNestedSetToFingerprint(fp, declaredIncludeSrcs);
-    fp.addInt(0); // mark the boundary between input types
     actionKeyContext.addNestedSetToFingerprint(fp, mandatoryInputs);
     fp.addInt(0);
     actionKeyContext.addNestedSetToFingerprint(fp, prunableHeaders);
